torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -36,17 +62,24 @@ class AttributePropagator {
     }
   }
 
-  void run(std::shared_ptr<Graph>& graph) {
+  void run() {
     auto applyInline = [](std::shared_ptr<Graph>& subgraph) {
       Inline(*subgraph);
     };
     auto applyOptimizations = [](std::shared_ptr<Graph>& subgraph) {
       runOptimization(subgraph, /* unroll? */ false);
     };
-    optimizeSubGraphs(graph, applyInline);
-    propagateAttributes(graph);
-    optimizeSubGraphs(graph, applyOptimizations);
-    cleanupFrozenModule(graph);
+    for (auto function : preservedMethods_) {
+      auto graph = function->graph();
+      optimizeSubGraphs(graph, applyInline);
```

Comment:
This is sequentially runs `propagateAttributes` on each method, which records mutable attributes and then inlines the non-mutable ones. What we want instead is to first collect the set of all mutable attributes on all methods, and then inline them.